modules/account-lib/src/coin/sol/utils.ts

```diff
@@ -196,6 +199,14 @@ export function getTransactionType(transaction: SolTransaction): TransactionType
       ValidInstructionTypesEnum.InitializeNonceAccount
   ) {
     return TransactionType.WalletInitialization;
+  } else if (
+    instructions.length <= 3 &&
+    getInstructionType(instructions[stakingActivateInstructionsIndexes.Initialize]) ===
+      ValidInstructionTypesEnum.StakingInitialize &&
+    getInstructionType(instructions[stakingActivateInstructionsIndexes.Delegate]) ===
+      ValidInstructionTypesEnum.StakingDelegate
+  ) {
+    return TransactionType.StakingActivate;
   } else {
     return TransactionType.Send;
   }
```

Comment:
This method is kinda of weird, Im open to suggestions since we are implementing two more builders

Reply:
you don't need the `else`, because the content of "if" has a `return`